mycore-ocfl/src/main/java/org/mycore/ocfl/MCROCFLAdaptionRepositoryProvider.java

```diff
@@ -0,0 +1,283 @@
+import edu.wisc.library.ocfl.core.extension.storage.layout.config.HashedNTupleIdEncapsulationLayoutConfig;
+
+/**
+ * The Adaptation Repository allows conversation between different OCFL Layout Types
+ * @author Tobias Lenhardt [Hammer1279]
+ * @version 1.0
+ */
+public class MCROCFLAdaptionRepositoryProvider extends MCROCFLRepositoryProvider {
+
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    public final MCROCFLXMLMetadataManager manager;
+
+    // if anything ever extends this class,
+    // the configurations dont have to be loaded again and can be called from child classes
+    protected final OcflExtensionConfig hashedNTupleConfig = new HashedNTupleIdEncapsulationLayoutConfig();
+
+    protected final OcflExtensionConfig mcrLayoutConfig = new MCRLayoutConfig();
+
+    private Path repositoryRoot;
+
+    private Path mainRoot;
+
+    private Path exportDir;
+
+    private Path backupDir;
+
+    private Path workDir;
+
+    private String layout;
+
+    /**
+     * New repository with new layout, this is for rewriting only
+     * and should not be used as base repository in production
+     */
+    private MutableOcflRepository repository;
+
+    /**
+     * Currently loaded "old" repository
+     */
+    private OcflRepository prevRepository;
+
+    Thread cleanDir = new Thread(() -> {
+        if (repositoryRoot.equals(mainRoot)) {
+            return;
+        }
+        if (this.repositoryRoot.toFile().exists()) {
+            LOGGER.info("Cleaning Directory on Exit...");
+            if (this.repositoryRoot.getFileName().equals("ocfl-root")) {
+                return;
+            }
+            exportDir.toFile().delete();
+            try (Stream<Path> walker = Files.walk(this.repositoryRoot)) {
+                walker
+                    .sorted(Comparator.reverseOrder())
+                    .map(Path::toFile)
+                    .forEach(File::delete);
+                walker.close();
+            } catch (IOException e) {
+                throw new MCRException("Error while cleaning directory: ", e);
+            }
+        }
+    });
+
+    public MCROCFLAdaptionRepositoryProvider(String repositoryKey) {
+        manager = new MCROCFLXMLMetadataManager();
+        manager.setRepositoryKey(repositoryKey);
+        setRepositoryRoot(MCRConfiguration2.getStringOrThrow("MCR.OCFL.Repository.Adapt.RepositoryRoot"));
+        this.mainRoot = Paths.get(MCRConfiguration2.getStringOrThrow("MCR.OCFL.Repository.Main.RepositoryRoot"));
+        setExportDir(MCRConfiguration2.getStringOrThrow("MCR.OCFL.Repository.Adapt.ExportDir"));
+        this.backupDir = Paths.get(MCRConfiguration2.getStringOrThrow("MCR.OCFL.Repository.Adapt.BackupDir"));
+        setWorkDir(MCRConfiguration2.getStringOrThrow("MCR.OCFL.Repository.Adapt.WorkDir"));
+        this.layout = MCRConfiguration2.getStringOrThrow("MCR.OCFL.Repository.Adapt.Layout");
+    }
+
+    @Override
```

Comment:
Most of the Code already exists in MCRSimpleOcflRepositoryProvider, why not just extend that class.